### PR TITLE
verify-linkchecker.sh now exits with a non-zero return code if found invalid links

### PR DESCRIPTION
**Pull request by caesarxuchao**
With this fix Jenkins will send email to me if linkchecker finds invalid links.

### CODE REVIEW

File: hack/after-build/verify-linkcheck.sh [hunk 1]
```diff
@@ -52,8 +52,7 @@ done
 if [ ${found_invalid} = true ]; then
   echo "Summary of invalid links:"
   cat ${OUTPUT}/error
+  exit 1
 fi
 
-trap "cleanup" EXIT SIGINT
```
Comment (caesarxuchao) [inline hack/after-build/verify-linkcheck.sh]: This is a duplication of line 32.

**Comment by caesarxuchao** [comment]
@k8s-bot e2e test this, issue: #21836

**Comment by ixdy** [comment]
LGTM

**Comment by caesarxuchao** [comment]
I'm going to manually merge it. It doesn't affect the production code.

**Pull request merged**